### PR TITLE
APPS-579 Add regression tests for expression evaluation: boolean, substitution.

### PR DESCRIPTION
**Pull request by r-i-v-a**

### CODE REVIEW

File: src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version 1.1
```
Comment (r-i-v-a) [inline src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl:1]: @jdidion  Maybe it should be a test against 1.0 instead?

Comment (r-i-v-a) [reply src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl:1]: Or should it be tested under syntax like https://github.com/dnanexus-rnd/wdlTools/pull/135/files ? They're also in dxCompiler integration tests in my other PR, here I just want to check that they compile without errors

Comment (jdidion) [reply src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl:1]: 1.1 is fine

**Review by jdidion — APPROVED** [review]

File: src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl (hunk 1 above)
Comment (r-i-v-a) [reply src/test/resources/eval/v1.1/apps_579_boolean_flag_expr.wdl:1]: @jdidion   I'm trying to add the test now. I only added the WDL files for the test so far.

File: src/test/scala/wdlTools/eval/EvalTest.scala [hunk 2]
```diff
@@ -609,4 +609,14 @@ class EvalTest extends AnyFlatSpec with Matchers with Inside {
     val outputExpr = task.outputs.head.expr
     evaluator.applyExpr(outputExpr) shouldBe V_String("1^I1$\n2\\t2$\n3\\^I3$\n4\\\\t4$")
   }
+
+  it should "evaluate command section with boolean flag expression" in {
```
[the remark this answers is not in the document]
Comment (r-i-v-a) [reply src/test/scala/wdlTools/eval/EvalTest.scala:612]: @jdidion Based on this result, maybe evalCommand() is the wrong type of test. It looks like it's not picking up the identifiers that were introduced in the input block https://github.com/dnanexus-rnd/wdlTools/pull/136/checks?check_run_id=2528543793

[the remark this answers is not in the document]
Comment (jdidion) [reply src/test/scala/wdlTools/eval/EvalTest.scala:612]: evalCommand doesn't evaluate the input block. You need to manually pass in the bindings. Look at some of the earlier tests for examples (the ones that come before the evalCommand function).

If the test requires full evaluation of the task, then that is a test of execution, not evaluation. I'd put that kind of test in dxCompiler instead.

[the remark this answers is not in the document]
Comment (r-i-v-a) [reply src/test/scala/wdlTools/eval/EvalTest.scala:612]: @jdidion   Thanks, have got the tests working now. This is ready for review.

**Review by jdidion — APPROVED** [review]

**Pull request merged**